calculator/src/App.js

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import "./App.css";
+
+class App extends React.Component {
+  state = { input: "", previousNum: "", currentNum: "", operator: "" };
+
+  handleClick = (e) => {
+    this.setState({
+      input: this.state.input + e.target.value,
+    });
+  };
+  // clears input when clicked
+  handleClear = (e) => {
+    this.setState({ input: "" });
+  };
+
+  handlePlusNegative = (e) => {
+    this.setState({
+      input: Number(this.state.input) * -1,
+    });
+  };
+
+  add = () => {
+    this.setState({
+      previousNum: this.state.input,
+      input: "",
+      operator: "plus",
+    });
+  };
+
+  handleEqual = () => {
+    // console.log(this.state.input);
+    this.state.currentNum = this.state.input;
```

Comment:
Illegal update of state. We must Never update state with direct assignment. When wanting to update or change we must Always use setState